### PR TITLE
Use sha256 instead of md5 for file integrity checks

### LINKED ISSUE #7029: Consider using sha256 instead of md5 in config/manifest.inc.php
Hi

md5 is officially broken(1) and we should consider migrating to sha256 in config/manifest.inc.php 

Thanks

Links:
1. http://en.wikipedia.org/wiki/Collision_attack#cite_note-2

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

Using sh256 instead of md5 will make the integrity check slightly more secure.

fixes #7029

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: core/FileIntegrity.php [hunk 1]
```diff
@@ -332,9 +332,9 @@ protected static function isFileNotInManifestButExpectedAnyway($file)
     protected static function getMessagesFilesMismatch($messages)
     {
         $messagesMismatch = array();
-        $hasMd5file = function_exists('md5_file');
+        $hasHashFile = function_exists('hash_file');
         $files = \Piwik\Manifest::$files;
-        $hasMd5 = function_exists('md5');
+        $hasHash = function_exists('hash');
```
Comment (mneudert) [inline core/FileIntegrity.php]: I would assume we can safely remove this check. The function is already used in many places without a check.

File: core/FileIntegrity.php [hunk 2]
```diff
@@ -332,9 +332,9 @@ protected static function isFileNotInManifestButExpectedAnyway($file)
     protected static function getMessagesFilesMismatch($messages)
     {
         $messagesMismatch = array();
-        $hasMd5file = function_exists('md5_file');
+        $hasHashFile = function_exists('hash_file');
```
Comment (mneudert) [inline core/FileIntegrity.php:335]: The `RecommendedFunctionsCheck` is testing for `md5_file`, should we add `hash_file` to the list in `getRecommendedFunctions` to match the other one?

Comment (sgiehl) [reply core/FileIntegrity.php:335]: yes, makes sense.

Comment (mneudert) [reply core/FileIntegrity.php:335]: Now we only need two fresh screenshots for the affected UI tests :eyes:

**Review by mneudert — APPROVED** [review]

**Pull request merged**